### PR TITLE
Add JSONAdapted classes for Appointment, Date, From, and To

### PR DESCRIPTION
**Pull request by wang-h-z**
This patch introduces four essential JSONAdapted classes:
- JSONAdaptedAppointment
- JSONAdaptedDate
- JSONAdaptedFrom
- JSONAdaptedTo

These classes facilitate the storage and deserialization of the
`Appointment` model, including its coupled classes `Date`, `From`,
and `To`. The implementation follows a similar structure to
`JsonAdaptedPerson` and `JsonAdaptedTag`, ensuring that the
deserialization process from JSON to the model classes occurs
seamlessly.

Additionally, updates were made to:
- `JsonAdaptedPersonTest` and `JsonSerializableAddressBookTest`
- `typicalPersonsAddressBook.json` and `duplicatePersonAddressBook.json`

These updates help to validate the new JSONAdapted classes and
ensure consistency across tests.

### CODE REVIEW

File: src/main/java/seedu/address/storage/JsonAdaptedPerson.java [hunk 1]
```diff
@@ -109,19 +100,11 @@ public Person toModelType() throws IllegalValueException {
...
-            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, Date.class.getSimpleName()));
-        }
-
-        if (from == null) {
-            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, From.class.getSimpleName()));
-        }
-
-        if (to == null) {
-            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, To.class.getSimpleName()));
+        if (appointment == null) {
```
Comment (lrongyi) [inline src/main/java/seedu/address/storage/JsonAdaptedPerson.java:103]: Good use of SLAP here, abstracting out the date, from, and to

File: src/test/data/JsonSerializableAddressBookTest/typicalPersonsAddressBook.json [hunk 2]
```diff
@@ -1,56 +1,96 @@
...
-    "tags" : [ ]
-  }, {
-    "name" : "George Best",
-    "phone" : "9482442",
-    "email" : "anna@example.com",
-    "address" : "4th street",
-    "remark" : "",
-    "tags" : [ ]
-  } ]
+  "_comment": "AddressBook save file containing typical persons from TypicalPersons#getTypicalAddressBook()",
```
Comment (lrongyi) [inline src/test/data/JsonSerializableAddressBookTest/typicalPersonsAddressBook.json:2]: LGTM, possible to remove the "remark" attribute from this file as well

File: src/test/data/JsonSerializableAddressBookTest/duplicatePersonAddressBook.json [hunk 3]
```diff
@@ -1,22 +1,29 @@
...
-    "name": "Alice Pauline",
-    "phone": "94351253",
-    "email": "pauline@example.com",
-    "property": "4th street",
-    "remark": "",
-    "date": "",
-    "from": "",
-    "to" : ""
-  } ]
+  "persons": [
```
Comment (lrongyi) [inline src/test/data/JsonSerializableAddressBookTest/duplicatePersonAddressBook.json:2]: LGTM, possible to remove the "remark" attribute

File: src/test/java/seedu/address/storage/JsonAdaptedAppointmentTest.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
+package seedu.address.storage;
```
Comment (lrongyi) [inline src/test/java/seedu/address/storage/JsonAdaptedAppointmentTest.java:1]: LGTM, need to include tests for JSONAdaptedDate/JSONAdaptedFrom/JSONAdaptedTo

Comment (wang-h-z) [reply src/test/java/seedu/address/storage/JsonAdaptedAppointmentTest.java:1]: Included in the newest commit. Please take a look.

**Pull request merged**